### DISCUSSION TITLE
how to perform navigation

### DISCUSSION THREAD

**Discussion by nitinss17** [TomSchimansky/CustomTkinter#1962] [Q&A] [answered]
Hello, 

I have a login page in login_page.py and then when they login successfully, I want to destroy the login page and open another main page which is written in another python script, and in this page when they click logout I want to destroy the main page and redirect them to login page. I do not want to merge all pages to one single python script because I have multiple pages in different python files. how can I perform navigation from one script to another? 

note : 
1. I don't want to withdraw and deiconify because this will consume memory, destroy would be a better option 
2. I cannot import both files in each other scripts because then it would create a circular reference
3. also I have different screen size and themes for login and main page 

highly appreciate if someone could guide me preferably programmatically. that would help a lot

Awaiting your kind response.

**Comment by ElectricCandlelight** [comment, marked as answer]
Why can't you have one file as entry point for the UI and handles state. Then the other files just for the functions. 
If it's just a log in page and menu page then the main entry point file wouldn't be that big. 

If you really want to have the UI that relates to the functions just have them as a class in each file and import them to the main entry point.

**Reply by nitinss17** [reply]
ok so ive implemented a navigation page class which is withdrawn in init and launch login class in another script, on login button click, login page is withdrawn and main page is shown and when logout is clicked the main page is destroyed and login page is deiconify() ... this seem to work. 
however i see a lot of flickering happening when i withdraw and deiconify, ill look into this, thank you for the suggestion

**Comment by DragonOfShuu** [comment]
If this were my project, I would have a base window class that manages logins and application resources. When the application first opens, it would open the login page. The user would login, and the information would get sent to this base window, and then the window would destroy the login and attach the main UI to itself. 

You can do this by passing callback functions to the pages from the main window, and then having the pages run these functions when they want to send information to the main window or destroy themselves and request a different page to open.

If you have ever written in something like ReactJS, these kinds of callbacks in and out of components should be familiar to you (of course if you were actually using ReactJS, how you would actually implement page switching would be different lol).
